PR#1 of issue #2

youcefs21 opened the pull request:
## Overview
Implement Assignment Enrollment

This change is a
- [x] New feature (non-breaking change which adds functionality)


## Description
- Fixed up the `AssignmentEnrollment` entity
- Created `AssignmentEnrollmentRepository`
- Created `AssignmentEnrollmentPage` model
- added test data to `bootstrap_data.sql`


## Motivation/Links
#2 

## How was this tested?
The project runs without crashing, and the `AssignmentEnrollment` table shows up on H2 with the test data I added to `bootstrap_data.sql`


## Todos
The rest of #2

youcefs21 committed:
PR#1 of issue #2

- Fixed up the `AssignmentEnrollment` entity
- Created `AssignmentEnrollmentRepository`
- Created `AssignmentEnrollmentPage` model
- added test data to `bootstrap_data.sql`

benjaminkostiuk commented on src/main/java/com/unityTest/courseManagement/entity/AssignmentEnrollment.java:
```diff
@@ -30,10 +34,10 @@
+	@ApiModelProperty(value = "Enrolled assignment")
+	@ManyToOne(fetch = FetchType.LAZY)
+	@JoinColumn(name = "ASSIGNMENT_ID", referencedColumnName = "ID")
+	private Assignment course;
```
I forgot to rename this, oops
```suggestion
	private Assignment assignment;
```

benjaminkostiuk commented on src/main/java/com/unityTest/courseManagement/entity/AssignmentEnrollment.java:
```diff
@@ -30,10 +34,10 @@
 	private int id;
```
Add a `@JsonIgnore` annotation to this id field. We don't need the ids returned by the endpoint.

benjaminkostiuk commented on src/main/resources/db/h2/bootstrap_data.sql:
```diff
@@ -63,3 +63,11 @@ VALUES
 (2, 'CASE', 1000, 'X02', 'How about being positive? :smile:', 0),
 
 (3, 'CASE', 1001, 'X03', 'Should we have done this instead? `x = 3`?', 1);
+
```
Could you add a couple rows to make it easy to enroll the current user you're testing with in assignments. (This will help when you're testing the endpoints since you'll only be able to pull for the current user)

Add this code at the top of the file
```
-- Replace this id with your test user
SET @USER_ID = 'TEST USER ID'
```
And then enroll them
```
(6, 1, @USER_ID, false),
(7, 2, @USER_ID, true);
```

You can find your user id in keycloak for when you're testing endpoints later.

benjaminkostiuk reviewed (changes requested):
Great stuff 👌! Couple small changes that I forgot when I first worked on the class but really good 👍

youcefs21 committed:
Fixed some things for PR#1 of issue #2

- Renamed the assignment associated with AssignmentEnrollment from "course" to "assignment
- Added a JsonIgnore tag to the ID in AssignmentEnrollment
- Added a USER_ID variable to bootstrap_data.sql for ease of testing
- Added a 2 more test cases to Assignment enrolment in bootstrap_data

benjaminkostiuk reviewed (approved):
Sweet I'll merge this right away :)
